sources/Algo/include/Algo.h

```diff
@@ -393,5 +398,300 @@ class ControllerInterfaceDefinitionAlgorithm : public NodeAlgorithm {
+   */
+  void dispatchAutomatonMulti(const inputs::AssemblingXmlDocument::DynamicAutomaton& automaton, const inputs::AssemblingXmlDocument::MacroConnect& macro,
+                              const inputs::AssemblingXmlDocument::MultipleAssociation& multiassoc);
+
+  /**
+   * @brief Process node in case of automaton bus connection
+   * @param node node to process
+   */
+  void processDynModelBusConnection(const NodePtr& node);
+
+  /**
+   * @brief Process node in case of automaton shunt connection
+   * @param node node to process
+   */
+  void processDynModelShuntConnection(const NodePtr& node);
+
+  /**
+   * @brief Process node in case of automaton line connection
+   * @param line line to process
+   */
+  void processDynModelLineConnection(const std::shared_ptr<inputs::Line>& line);
+
+  /**
+   * @brief Process node in case of automaton transfo connection
+   * @param tfo transfo to process
+   */
+  void processDynModelTfoConnection(const std::shared_ptr<inputs::Tfo>& tfo);
+
+  /**
+   * @brief Add macro connection to node connections of the automaton definition
+   *
+   * Creates the dynamic model definition if not already existing
+   * Update the dynamic model definitions member
+   *
+   * @param automaton the config automaton
+   * @param macroConnection the macro connection to add
+   */
+  void addMacroConnectionToDef(const dfl::inputs::AssemblingXmlDocument::DynamicAutomaton& automaton,
+                               const DynModelDefinition::MacroConnection& macroConnection);
+
+ private:
+  DynModelDefinitions& models_;  ///< automaton definitions to update
+
+  std::unordered_map<std::string, inputs::AssemblingXmlDocument::DynamicAutomaton> automatonsById_;  ///< config automaton by model id
+  std::unordered_map<inputs::VoltageLevel::VoltageLevelId, std::unordered_set<MacroConnect, MacroConnectHash>>
+      macroConnectByVlForBusesId_;  ///< macro connections for buses, by voltage level
+  std::unordered_map<inputs::VoltageLevel::VoltageLevelId, std::vector<MacroConnect>>
+      macroConnectByVlForShuntsId_;                                                             ///< macro connections for shunts, by voltage level
+  std::unordered_map<inputs::Line::LineId, std::vector<MacroConnect>> macroConnectByLineName_;  ///< macro connections for lines, by line id
+  std::unordered_map<inputs::Tfo::TfoId, std::vector<MacroConnect>> macroConnectByTfoName_;     ///< macro connections for transfo, by transfo id
+
+  const inputs::DynamicDataBaseManager& manager_;  ///< dynamic database config manager
+};
+
+/**
+ * @brief Shunt counter definition
+ */
+struct ShuntCounterDefinitions {
+  std::unordered_map<inputs::VoltageLevel::VoltageLevelId, unsigned int> nbShunts;  ///< Number of shunts by voltage level
+};
+
+/**
+ * @brief Counter of shunts by voltage levels
+ */
+class ShuntCounterAlgorithm : public NodeAlgorithm {
+ public:
+  /**
+   * @brief Constructor
+   * @param defs the counter definitions to update
```

Comment:
shuntCounterDefs (same goes for attribute)

Reply:
OK